### PR TITLE
Speedmeter OK

### PR DESCRIPTION
**Pull request by hensldm**
Mostly the same as OoT, but with added support for high resolution.
Docs come from OoT, thank you :) .
A couple fakes unfortunately.

### CODE REVIEW

File: src/code/speed_meter.c
Comment (hensldm) [reply src/code/speed_meter.c:232]: From what I can tell yeah. Otherwise bad regalloc.

**Review by AngheloAlf — APPROVED** [review]

File: include/z64speed_meter.h
Comment (engineer124) [inline include/z64speed_meter.h:27]: Is this unused? It seems very obvious as to it's use by surrounding variables, but would we still want to name this `sPad`?

File: src/code/speed_meter.c
Comment (engineer124) [inline src/code/speed_meter.c:87]: These two alt macros are a bit annoying :( should this one go in z64view.h?

File: src/code/speed_meter.c
Comment (engineer124) [inline src/code/speed_meter.c:173]:
```suggestion
    this->maxVal = maxVal;
```
Seems like 2 words?

File: src/code/speed_meter.c
Comment (engineer124) [inline src/code/speed_meter.c:242]: += 2 doesn't work?

File: include/z64speed_meter.h
Comment (hensldm) [reply include/z64speed_meter.h:27]: The only issue is I would very much prefer externing it in the header so I don't have to deal with bss as much. Perhaps we could name it something like `gUnkTimeAcc` to give it a unique name and can extern it without issue.

File: src/code/speed_meter.c
Comment (hensldm) [reply src/code/speed_meter.c:87]: Hmm, now that I think about it yeah. `SET_FULLSCREEN_VIEWPORT_ALT` is actually meant to be used for dynamic fullscreen (i.e. it can support both hires and lores). Tbh, I kind of want to name this without the `ALT` and change the other one to `SET_FULLSCREEN_VIEWPORT_LORES` since it only support lores. Thoughts?

File: src/code/speed_meter.c
Comment (hensldm) [reply src/code/speed_meter.c:242]: It actually does not :(

File: src/code/speed_meter.c
Comment (hensldm) [reply src/code/speed_meter.c:87]: The other option is name `SET_FULLSCREEN_VIEWPORT_ALT` `SET_FULLSCREEN_VIEWPORT_HIRES` since it supports hires.

Comment (engineer124) [reply src/code/speed_meter.c]: So the main difference between `SET_FULLSCREEN_VIEWPORT_HIRES` and `SET_FULLSCREEN_VIEWPORT` is `gScreenHeight` vs `SCREEN_HEIGHT`? Could `gScreenHeight` be either lores or hires? I'm not sure how `gScreenHeight` actually works in practice (and in parallel `gScreenWidth`)

Comment (hensldm) [reply src/code/speed_meter.c]: Correct that is the only difference. When LoRes or HiRes gets set through sys_cfb, both `gScreenHeight` and `gScreenWidth` get updated to the appropriate values (see `SysCfb_SetLoResMode` and `SysCfb_SetHiResMode`.

Comment (engineer124) [reply src/code/speed_meter.c]: I'm just thinking, because `SET_FULLSCREEN_VIEWPORT_HIRES` could be either HiRes or LowRes right? It's not just limited to HiRes as the name implies?

Comment (hensldm) [reply src/code/speed_meter.c]: Yeah hence we could go with the first option (rename the other one to `SET_FULLSCREEN_VIEWPORT_LORES`). I chose the second option mainly because the `SET_FULLSCREEN_VIEWPORT_LORES` is used so much since the game more often than not doesn't care about lores vs hires (only bombers notebook I think), and it matches OoT better. If you prefer I can go with that option though.

Comment (engineer124) [reply src/code/speed_meter.c]: Ah sorry that's not what I meant. The other one can be left alone. I mean you can use `SET_FULLSCREEN_VIEWPORT_HIRES` and it would set it to LoRes, which seems misleading.

**Review by engineer124 — APPROVED** [review]

File: src/code/speed_meter.c
Comment (tom-overton) [inline src/code/speed_meter.c:200]: Do you have a decomp.me for this?

**Review by tom-overton — APPROVED** [review]
Just one tiny thing, but I'll approve anyways.

File: src/code/speed_meter.c
Comment (hensldm) [reply src/code/speed_meter.c:200]: https://decomp.me/scratch/mstcw

Comment (tom-overton) [reply src/code/speed_meter.c:200]: Ah, sorry, I should've clarified. I try to put decomp.me links below `//! FAKE` comments so people can attempt to fix them. However, in this case, it's not super necessary, since you could just delete the following line and try it for yourself. So I'll leave it up to you whether you want to put that in or not.

Comment (hensldm) [reply src/code/speed_meter.c:200]: Ah, tbh working locally with this is probably best anyways, due to the graphic macros.

Comment (tom-overton) [reply src/code/speed_meter.c:200]: Yeah, fair enough. No need to change anything then; my approval stands with the code as-is.

**Pull request merged**
